kernels/typescript/default.nix

```
@@ -6,21 +6,10 @@
   inherit (pkgs) fetchurl lib stdenv writeScriptBin;
   inherit (lib) makeBinPath;
 
-  # This is honestly black magic and I forget where/how I discovered it.
-  tslab = let
-    inherit (lib) composeManyExtensions extends makeExtensible mapAttrs;
-
-    nodePackages = final:
-      import ./composition.nix {
-        inherit pkgs nodejs;
-        inherit (stdenv.hostPlatform) system;
-      };
-
-    extensions = composeManyExtensions [
-      (import ./overrides.nix {inherit pkgs nodejs;})
-    ];
-  in
-    (makeExtensible (extends extensions nodePackages)).tslab;
+  tslab = pkgs.npmlock2nix.build {
```

We should probably use 
```suggestion
  tslab = pkgs.npmlock2nix.node_modules {
```
Since we are not really building anything in this project.

Executable will then be in `$out/node_modules/.bin` folder.

---

I tried building the tslab project and it doens't show up in the `.bin` directory.

```shell
$ ls -l result/bin/
total 33
lrwxrwxrwx 1 root root 18 Dec 31  1969 acorn -> ../acorn/bin/acorn
lrwxrwxrwx 1 root root 19 Dec 31  1969 atob -> ../atob/bin/atob.js
lrwxrwxrwx 1 root root 22 Dec 31  1969 browserslist -> ../browserslist/cli.js
lrwxrwxrwx 1 root root 32 Dec 31  1969 browserslist-lint -> ../update-browserslist-db/cli.js
lrwxrwxrwx 1 root root 29 Dec 31  1969 escodegen -> ../escodegen/bin/escodegen.js
lrwxrwxrwx 1 root root 30 Dec 31  1969 esgenerate -> ../escodegen/bin/esgenerate.js
lrwxrwxrwx 1 root root 25 Dec 31  1969 esparse -> ../esprima/bin/esparse.js
lrwxrwxrwx 1 root root 28 Dec 31  1969 esvalidate -> ../esprima/bin/esvalidate.js
lrwxrwxrwx 1 root root 31 Dec 31  1969 import-local-fixture -> ../import-local/fixtures/cli.js
lrwxrwxrwx 1 root root 15 Dec 31  1969 is-ci -> ../is-ci/bin.js
lrwxrwxrwx 1 root root 19 Dec 31  1969 is-docker -> ../is-docker/cli.js
lrwxrwxrwx 1 root root 19 Dec 31  1969 jest -> ../jest/bin/jest.js
lrwxrwxrwx 1 root root 35 Dec 31  1969 jest-runtime -> ../jest-runtime/bin/jest-runtime.js
lrwxrwxrwx 1 root root 18 Dec 31  1969 jsesc -> ../jsesc/bin/jsesc
lrwxrwxrwx 1 root root 19 Dec 31  1969 json5 -> ../json5/lib/cli.js
lrwxrwxrwx 1 root root 25 Dec 31  1969 js-yaml -> ../js-yaml/bin/js-yaml.js
lrwxrwxrwx 1 root root 20 Dec 31  1969 mkdirp -> ../mkdirp/bin/cmd.js
lrwxrwxrwx 1 root root 24 Dec 31  1969 node-gyp-build -> ../node-gyp-build/bin.js
lrwxrwxrwx 1 root root 29 Dec 31  1969 node-gyp-build-optional -> ../node-gyp-build/optional.js
lrwxrwxrwx 1 root root 31 Dec 31  1969 node-gyp-build-test -> ../node-gyp-build/build-test.js
lrwxrwxrwx 1 root root 23 Dec 31  1969 node-which -> ../which/bin/node-which
lrwxrwxrwx 1 root root 36 Dec 31  1969 parser -> ../@babel/parser/bin/babel-parser.js
lrwxrwxrwx 1 root root 27 Dec 31  1969 prettier -> ../prettier/bin-prettier.js
lrwxrwxrwx 1 root root 22 Dec 31  1969 resolve -> ../resolve/bin/resolve
lrwxrwxrwx 1 root root 16 Dec 31  1969 rimraf -> ../rimraf/bin.js
lrwxrwxrwx 1 root root 25 Dec 31  1969 rollup -> ../rollup/dist/bin/rollup
lrwxrwxrwx 1 root root 18 Dec 31  1969 sane -> ../sane/src/cli.js
lrwxrwxrwx 1 root root 23 Dec 31  1969 semver -> ../semver/bin/semver.js
lrwxrwxrwx 1 root root 21 Dec 31  1969 tsc -> ../typescript/bin/tsc
lrwxrwxrwx 1 root root 17 Dec 31  1969 ts-jest -> ../ts-jest/cli.js
lrwxrwxrwx 1 root root 26 Dec 31  1969 tsserver -> ../typescript/bin/tsserver
lrwxrwxrwx 1 root root 21 Dec 31  1969 uuid -> ../uuid/dist/bin/uuid
lrwxrwxrwx 1 root root 26 Dec 31  1969 watch -> ../@cnakazawa/watch/cli.js
```

---

oh `bin -> node_modules/.bin/`

---

I spoke with @Radvendii about this and `npmlock2nix.build` is the right way to go.